File: c++代码/226.翻转二叉树.cpp
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution
{
public:
    TreeNode *invertTree(TreeNode *root)
    {
        if (!root)
            return NULL;
        helper(root);
        return root;
    }

    void helper(TreeNode *root)
    {
        if (root->left)
            helper(root->left);
        if (root->right)
            helper(root->right);
        TreeNode *tmp = root->left;
        root->left = root->right;
        root->right = tmp;
    }
};

File: 力扣/每日一题/226. 翻转二叉树.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if not root: return None
        def helper(root):
            if root.left: helper(root.left)
            if root.right: helper(root.right)
            tmp = root.left
            root.left = root.right
            root.right = tmp
        helper(root)
        return root
